x-pack/plugins/lens/public/indexpattern_datasource/operations/definitions/formula/validation.ts

```diff
@@ -633,6 +701,22 @@ function runFullASTValidation(
             })
           );
         } else {
+          const fields = variables.filter(
```

Comment:
Which case is caught here?

Reply:
Probably need to add some more comment context on all these branches.
In this specific branch it is checking that only a single `field` is passed to `field`-only operations.

Reply:
I don't think so, this branch is in the `if (nodeOperation.input === 'fullReference')` block.

I don't see in which case it would become necessary - the only one is this, but it shows two error for the same problem:
<img width="708" alt="Screenshot 2021-07-06 at 11 55 05" src="https://user-images.githubusercontent.com/1508364/124581179-2803ff80-de51-11eb-9752-456b2fb6ab3a.png">

Reply:
You're right sorry. The check has to exclude the first argument.
This check is here to validate this scenario:

```
moving_average(average(bytes), bytes)
```

I'll refactor the logic to group both first argument check and all arguents check for field values.